File: src/main/java/com/pepebot/model/TradeLog.java
package com.pepebot.model;

import java.math.BigDecimal;
import java.time.Instant;

import com.pepebot.dto.OrderType;
import com.pepebot.dto.Side;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "trade_log")
public class TradeLog {

	@Id
	private String id;

	@Column(nullable = false)
	private Instant executedAt;

	@Column(nullable = false)
	private String symbol;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	private Side side;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	private OrderType type;

	@Column(nullable = false, precision = 38, scale = 18)
	private BigDecimal volume;

	@Column(nullable = false, precision = 38, scale = 18)
	private BigDecimal price;

	@Column(nullable = false)
	private String status;

	@Column(nullable = false)
	private String orderId;

	@Column(nullable = false)
	private String message;

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public Instant getExecutedAt() {
		return executedAt;
	}

	public void setExecutedAt(Instant executedAt) {
		this.executedAt = executedAt;
	}

	public String getSymbol() {
		return symbol;
	}

	public void setSymbol(String symbol) {
		this.symbol = symbol;
	}

	public Side getSide() {
		return side;
	}

	public void setSide(Side side) {
		this.side = side;
	}

	public OrderType getType() {
		return type;
	}

	public void setType(OrderType type) {
		this.type = type;
	}

	public BigDecimal getVolume() {
		return volume;
	}

	public void setVolume(BigDecimal volume) {
		this.volume = volume;
	}

	public BigDecimal getPrice() {
		return price;
	}

	public void setPrice(BigDecimal price) {
		this.price = price;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public String getOrderId() {
		return orderId;
	}

	public void setOrderId(String orderId) {
		this.orderId = orderId;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}
}

File: src/main/kotlin/com/pepebot/model/TradeLog.kt
package com.pepebot.model

import com.pepebot.dto.OrderType
import com.pepebot.dto.Side
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal
import java.time.Instant

@Entity
@Table(name = "trade_log")
open class TradeLog {
    @Id
    open var id: String = ""

    @Column(nullable = false)
    open var executedAt: Instant = Instant.EPOCH

    @Column(nullable = false)
    open var symbol: String = ""

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    open var side: Side = Side.BUY

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    open var type: OrderType = OrderType.MARKET

    @Column(nullable = false, precision = 38, scale = 18)
    open var volume: BigDecimal = BigDecimal.ZERO

    @Column(nullable = false, precision = 38, scale = 18)
    open var price: BigDecimal = BigDecimal.ZERO

    @Column(nullable = false)
    open var status: String = ""

    @Column(nullable = false)
    open var orderId: String = ""

    @Column(nullable = false)
    open var message: String = ""
}
